src/js/components/DataTable/Header.js

```diff
@@ -183,6 +191,15 @@ const Header = ({
               }
             } else content = header;
 
+            if (unitsContent) {
+              content = (
+                <Box {...theme.dataTable.header.units.container}>
```

Comment:
Something feels a little odd to have `units.container` driving the Box containing the main header content.

I'm also wondering how this interacts with `columns.align`, since that also has some layout aspects to it.

With the way this PR is structured, the sort icon will be placed between the heading label and the units. I'm wondering if that's the typical design approach. I could see designs wanting the units to be next to the label with the sort icon following the units.

Here are some rough possible designs:
| label units            |
|            label units |
| label            units |
| label units sort    |
|    label units sort |
| label sort    units |

At this point, I'm wondering if the last design should require a custom column.header render function and that the theme should be more focused around just the Text for the units, so the theme could have just a color, size, margin.

Reply:
I think part of my confusion is around the naming of "units.container" compared to what it is affecting.

Would it work to just do: `<Box align="baseline">{content}<Text {...theme.dataTable.units} /></Box>` when there are units and put a left margin on the Text?

Reply:
That's certainly possible if that works for you (+ direction="row"). The nice part about having all the box props in the theme under something like container is it gives you full control rather than having the hardcoded parts (kinda Shimi's earlier comment.) Although most of the other layout in Header.js is hardcoded.

Reply:
If we decide to keep it, I think we would need to put it somewhere other than under `units`. For example, if I set `units.container.background = 'brand'`, it will be the background of both the label and the units. Something on the order of `theme.dataTable.labelUnitsContainer` would be more accurate.

However, I tend towards not putting it in the theme at the moment. As you mention, there's a bunch of other styling stuff hardcoded in the Header already. I'd rather bring it all into the theme at the same time, driven by some specific design references.

Reply:
Simplified...Does this look a little better?

Reply:
yep